packages/core/src/shared/utilities/pollingSet.ts

```diff
@@ -0,0 +1,52 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { globals } from '..'
+
+/**
+ * A useful abstraction that does the following:
+ * - keep a set of items.
+ * - if the set is non-empty, run some action every interval seconds.
+ * - once the set empties, clear the timer
+ * @param interval the interval in seconds
+ * @param action the action to perform
+ */
+export class PollingSet<T> extends Set<T> {
```

Comment:
This reminds me of Go's `WaitGroup` (which I've also seen in other nodejs projects). https://pkg.go.dev/sync#WaitGroup

Not sure if it's exactly the same, but might be worth linking to https://pkg.go.dev/sync#WaitGroup from the docstring here for reference, in case we also need that concept in the future.